### PR TITLE
recoll: refactor and 1.24.5 -> 1.27.12

### PR DESCRIPTION
**Pull request by fabianhjr**
Changes:

- add pkg-config
- name -> pname + version
- python2 -> python3
- patch /usr/share/recoll to $out/share/recoll
- remove Makefile patch for macOS that has been merged upstream (
  https://framagit.org/medoc92/recoll/-/commit/b5c5734d064c638b702601fd391037a35b5bb2a1 )

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabianhjr** [comment]
There is still an issue pending on RECOLL_DATADIR

File: pkgs/applications/search/recoll/default.nix [hunk 1]
```diff
@@ -1,60 +1,88 @@
...
+        substituteInPlace $f --replace '"pstotext"'  '"${lib.getBin ghostscript}/bin/ps2ascii"'
+        substituteInPlace $f --replace '"sed"'       '"${lib.getBin gnused}/bin/sed"'
+        substituteInPlace $f --replace '"tar"'       '"${lib.getBin gnutar}/bin/tar"'
+        substituteInPlace $f --replace '"unzip"'     '"${lib.getBin unzip}/bin/unzip"'
+        substituteInPlace $f --replace '"xls2csv"'   '"${lib.getBin catdoc}/bin/xls2csv"'
+        substituteInPlace $f --replace '"xsltproc"'  '"${lib.getBin libxslt}/bin/xsltproc"'
+        substituteInPlace $f --replace '"unrtf"'     '"${lib.getBin unrtf}/bin/unrtf"'
+        substituteInPlace $f --replace '"untex"'     '"${lib.getBin untex}/bin/untex"'
+        substituteInPlace $f --replace '"wpd2html"'  '"${lib.getBin libwpd}/bin/wpd2html"'
+        substituteInPlace $f --replace /usr/bin/perl   ${lib.getBin perl}/bin/perl
```
Comment (ryantm) [inline pkgs/applications/search/recoll/default.nix:85]: Deleted a bunch of whitespace, but then added 2 spaces :)

**Comment by ryantm** [comment]
@jcumming , maintainer, for review and testing

**Comment by ryantm** [comment]
Result of `nixpkgs-review pr 104699` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>recoll</li>
  </ul>
</details>

File: pkgs/applications/search/recoll/default.nix (hunk 1 above)
Comment (fabianhjr) [reply pkgs/applications/search/recoll/default.nix:85]: wanted to align everything DX

Can remove it if it is not something desirable.

**Comment by jcumming** [comment]
LGTM!

**Pull request merged**